[8.0.1.0.0]-[product_category_workflow]

azmimr67 opened the pull request:
Product Category Work Flow
Plus check validasi saat restart

azmimr67 committed:
[8.0.1.0.0]-[product_category_workflow]
Product Category Work Flow
Plus check validasi saat restart

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
```
Error E302.
Sebelum class harus dibuat jarak 2 baris.

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+    @api.depends(
+        "company_id",
+    )
+
```
Error E304.
Sehabis penulisan decorator, tidak usah dikasi baris/enter

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+    parent_id = fields.Many2one(
+        comodel_name="product.category",
+        readonly=True,
+        states={"draft": [("readonly", False),]},
```
Error E231.
Sehabis koma harus dikasih spasi 1

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+    )
+    name = fields.Char(
+        readonly=True,
+        states={"draft": [("readonly", False),]},
```
Error E231.
Sehabis koma harus dikasih spasi 1

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+    )
+    type = fields.Selection(
+        readonly=True,
+        states={"draft": [("readonly", False),]},
```
Error E231.
Sehabis koma harus dikasih spasi 1

mikevhe18 commented on product_category_workflow_policy/models/product_category.py:
```diff
@@ -0,0 +1,174 @@
+        for document in self:
+            # cek jika memiliki child muncul error
+            if document.child_id:
+                raise UserError(_("Category %s cannot be set to draft because has child category") % (self.name))
```
Untuk sementara, fungsi pengecekan child ini dihapus dulu aja om.
Sepertinya perlu didiskusikan lagi flownya.
Contoh Kasus:
1. Product Category A punya 100 Child, kalau untuk mengnonaktifkan Product Category A diperlukan mengnonaktifkan Childnya terlebih dahulu, maka User perlu mengnonaktifkan 100 Child baru Product Category A bisa dinonaktifkan.
2. Product Category A mau diganti namanya, maka secara flow yg sekarang harus di restart. Kalau ada pengecekan Child, maka untuk merename, User perlu mengnonaktifkan 100 Child baru Product Category A bisa direname.

mikevhe18 commented on product_category_workflow_policy/models/res_company.py:
```diff
@@ -0,0 +1,33 @@
+    product_category_confirm_grp_ids = fields.Many2many(
+        string="Allow To Confirm Category",
+        comodel_name="res.groups",
+        relation="rel_company_id_confirm",
```
Penamaan relasi diganti jadi **"rel_company_2_product_categ_confirm"**.
Harus dibuat seunik mungkin untuk mencegah ada nama relasi yang sama

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/models/res_company.py:
```diff
@@ -0,0 +1,33 @@
+    product_category_valid_grp_ids = fields.Many2many(
+        string="Allow To Validate Category",
+        comodel_name="res.groups",
+        relation="rel_company_id_valid",
```
Penamaan relasi diganti jadi **"rel_company_2_product_categ_valid"**.
Harus dibuat seunik mungkin untuk mencegah ada nama relasi yang sama

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/models/res_company.py:
```diff
@@ -0,0 +1,33 @@
+    product_category_restart_grp_ids = fields.Many2many(
+        string="Allow To Restart Category",
+        comodel_name="res.groups",
+        relation="rel_company_id_restart",
```
Penamaan relasi diganti jadi **"rel_company_2_product_categ_restart"**.
Harus dibuat seunik mungkin untuk mencegah ada nama relasi yang sama

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/views/product_category_view.xml:
```diff
@@ -0,0 +1,87 @@
+<openerp>
+    <data>
+        <record id="product_category_view_form" model="ir.ui.view">
+            <field name="name">product.category form</field>
```
diganti jadi **"product.category.workflow.policy.form"**

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/views/product_category_view.xml:
```diff
@@ -0,0 +1,87 @@
+            </field>
+        </record>
+        <record id="product_category_list_view_tree" model="ir.ui.view">
+            <field name="name">product.category.list tree</field>
```
diganti jadi **"product.category.workflow.policy.tree"**

azmimr67 replied:
done

mikevhe18 commented on product_category_workflow_policy/views/res_company_views.xml:
```diff
@@ -0,0 +1,29 @@
+<data>
+
+<record id="res_company_view_form" model="ir.ui.view">
+    <field name="name">res.company form</field>
```
diganti jadi **"res.company.product.category.workflow.policy.form"**

azmimr67 replied:
done

azmimr67 committed:
[8.0.1.0.0][product_category_workflow]
Revisi file product_category.py
Revisi res_partner_py product_category_view xml sesuai comment di github
